### PR TITLE
Add benchmark scripts

### PR DESCRIPTION
**Pull request by borzunov**
This PR:

- Adds benchmark scripts for inference, forward pass, and full training step (e.g. used for experiments in our paper).
- Fixes bug with dtypes in `petals.DistributedBloomForSequenceClassification`.
- (minor refactor) Moves `DTYPE_MAP` to `petals.constants` as a useful constant.

### CODE REVIEW

File: benchmarks/benchmark_forward.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+def benchmark_forward(process_idx, args):
+    tokenizer = BloomTokenizerFast.from_pretrained(args.model)
+    model = DistributedBloomForCausalLM.from_pretrained(
+        args.model, initial_peers=args.initial_peers, torch_dtype=torch.bfloat16
+    )
+    logger.info(f"Created model: {process_idx=} {model.device=}")
+
+    torch.manual_seed(42)
+    for step in range(args.n_steps):
+        input_ids = torch.randint(100, 10000, size=(args.batch_size, args.seq_len))
```
Comment (justheuristic) [inline benchmarks/benchmark_forward.py:48]: [nit] we can use vocab size from config instead of 10k

File: benchmarks/benchmark_training.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    for step in range(args.n_steps):
+        input_ids = torch.randint(100, 10000, size=(args.batch_size, args.seq_len), device=args.device)
+        if args.task == "cls":
+            labels = torch.randint(0, 2, size=[args.batch_size], device=args.device)
+        else:
+            labels = input_ids
+
+        logger.info(f"{process_idx=} {step=} Forward")
+        start_time = perf_counter()
+        outputs = model(input_ids, labels=labels)
```
Comment (justheuristic) [inline benchmarks/benchmark_training.py:74]: [nit] would be nice to explain the warmup batch, and, *optionally*, allow using multiple warmup batches

Comment (borzunov) [reply benchmarks/benchmark_training.py:80]: [done]

File: benchmarks/benchmark_forward.py (hunk 1 above)
Comment (borzunov) [reply benchmarks/benchmark_forward.py]: [done]

**Pull request merged**